mitodl/mitxonline
# New design: adds program page

# What are the relevant tickets?

Part of #1773 
https://github.com/mitodl/hq/issues/1947
https://github.com/mitodl/hq/issues/1544

# Description (What does it do?)

Adds the program page layout to the system, including the course selector. 

# Screenshots (if appropriate):

![program-page-full](https://github.com/mitodl/mitxonline/assets/945611/2bfe26b3-94d1-4ffa-a221-c0e13834af81)

Enroll Now before course selection:

![Screenshot 2023-09-07 at 2 52 02 PM](https://github.com/mitodl/mitxonline/assets/945611/2d23c59e-ed26-4351-aed6-d89c53336733)

Enroll Now after selection:

![Screenshot 2023-09-07 at 2 52 07 PM](https://github.com/mitodl/mitxonline/assets/945611/69f30818-bf4b-44b9-924c-2844faac9066)

# How can this be tested?

For this to work, you will need the `mitxonline-new-product-page` flag enabled, and ideally a program page that has been filled out. Note that program pages will be mostly empty unless you've put info in them - since we had the pages but weren't exposing them, there wasn't a need to have these ready with data.

Testing is simply visiting a program page. Program pages are linked to from course pages. All operations should work as expected and the desktop design should match the Figma layout.

Additionally:
- The course requirement breakdowns use the title of their respective nodes from the requirements tree. If they are empty strings, they will default to "Electives" and "Core Courses". 
- The price is always free. 
- The certificate track pricing information comes from the Price block on the Program page - we (still) don't allow you to purchase an upgrade for an entire program, so this is left for the admin to specify. 
- The enrollment screen hides the certificate track information until a course is selected (as there's no data for that otherwise). Selecting a course will pull in the certificate track information and financial assistance information for that course. 
- This does not automatically enroll you in the program since you'll get an enrollment when you choose and enroll in the course.

# Additional Context

There's a couple edge cases that I don't think were planned for:
- If all courses are audit-only, we still ask them to upgrade. This will also probably break the selector. I wasn't sure what to do here. (This is probably fine to leave alone, though, since I don't know that this is a valid state anyway.)
- Similarly, there's not a good option if the learner selects a program that doesn't have an upgrade path (i.e. most have upgrades, but one doesn't for whatever reason). 
- The start date is just the earliest start date of the courses that are there. 

Additionally:
- This fixes a regression in the sidebar CSS - the max-width of that part of the app should be 400px, not 350, and that's fixed in this PR. 
- The price display in the Course Enrollment is now fixed in this PR (it was not considering approved flexible pricing).
- This removes `font-face` directives in its CSS - it should use the defaults. (On my system, this means it did not pick up the new fonts we're using.)
